### DISCUSSION TITLE
Feed Manager changing all my dateAdded entries

### DISCUSSION THREAD

**Discussion by jsonnentag** [rrirower/Channel-Feed-Manager#170] [Q&A]
Is there some good reason why the Feed Manager thinks the dateAdded value has to be 1 day before the releaseDate entry value?

I have pretty much all of the dates exactly the same but it wants to change all the dateAdded values.  (???)

It also spits out a pile of errors saying the dates and times aren't valid, when the format came straight from the original DP feed example.

**Comment by rrirower** [comment]
Please provide an example of the error.

**Comment by jsonnentag** [comment]
Feed file is located here and example specified will by Short Form item 260 until that particular Category is updated on Friday, although it will be in exactly the same format, just different linked file and date.

[http://sabbathschooltv.com/roku/sabbathschooltv.json](url)

In Firefox here I see a listing like this so it is easy to see shortFormVideos item 260

![image](https://github.com/rrirower/Channel-Feed-Manager/assets/36777806/34ff76eb-80db-41ee-a4f8-86e00ce8b51f)


![image](https://github.com/rrirower/Channel-Feed-Manager/assets/36777806/09a5a1e3-6e65-417a-8c46-a682c7fd0639)



This is what I see when viewing it in Channel Feed Manager (first item in the top of the category list in Miscellaneous):

![image](https://github.com/rrirower/Channel-Feed-Manager/assets/36777806/27228e53-be23-4609-8dff-eb865a0ee42b)

**Reply by rrirower** [reply]
Can you provide the log file associated with the above?

**Reply by jsonnentag** [reply]
I cleared out the log file and re-did the same things a couple times and it doesn't say anything about changing the date, but it does each time.  I have to add a rating and rating source before it can be saved, but each time it is done the date is changed.  ShortForm entry 262 became this for the log file attached.

![image](https://github.com/rrirower/Channel-Feed-Manager/assets/36777806/76abb4f8-afd9-46b6-b73c-f0e53beb197b)
[2024-03-24ChannelFeedManager.json](https://github.com/rrirower/Channel-Feed-Manager/files/14736934/2024-03-24ChannelFeedManager.json)

**Comment by jsonnentag** [comment]
Looking at the listed feed specification I can see why something is deemed wrong with the added time, but I didn't make up the "Z" on the end of what I have for everything.  That (except for the hours at the beginning) was what was in their example of a functioning JSON file 6 years ago, so that is what everything I have ever created has in it.  :))

![image](https://github.com/rrirower/Channel-Feed-Manager/assets/36777806/11c15b7a-ffa2-4189-b362-f8182e0cc889)


I guess search and replace can fix all of those.

**Comment by jsonnentag** [comment]
Yeah!  Their example feed (linked to at the bottom of the JSON Feed Specification page) still has that in it so it ought not to be considered a problem!

[https://devtools.web.roku.com/samples/roku-developers-feed-v1-rdp.json](url)

![image](https://github.com/rrirower/Channel-Feed-Manager/assets/36777806/03ba4fd3-3ac9-457c-a36a-f8541da501e2)

**Comment by jsonnentag** [comment]
The only difference I see is that the example now doesn't show the time down to the thousandths of a second, but I didn't make that up all on my own either.  It was in some example they had.  :))

**Anyway, Channel Feed Manager changing my dateAdded dates ought NOT to be happening.**

**Comment by rrirower** [comment]
By any chance, do you have more than one feed file with similar names in the channel database?  For example, "Sabbath School1.json" and "Sabbath School2.json".  I used the latest feed file you referenced above and I don't see the error.  However, my code includes a fix for a bug I've been working on.

**Reply by jsonnentag** [reply]
No, there is only 1 feed file imported into the channel database.  Just to do everything again I downloaded the JSON file from the website and then imported it into the Feed file library (that I had deleted the previous version out of) and it still does the same thing (changing the date added date).

This is what the version downloaded a few days ago says it is:

![image](https://github.com/rrirower/Channel-Feed-Manager/assets/36777806/e1dd48c5-e061-46f9-b702-fa18f3ca793a)

**Comment by rrirower** [comment]
Here's what I get when opening "Adventist News Network - March 22, 2024".  I can't recreate your issue.  Can you provide the steps you use to recreate the issue, or, a screen capture video that shows it?

![Dates](https://github.com/rrirower/Channel-Feed-Manager/assets/34550407/e21a747c-3078-49d5-b125-eaa2ba79825d)

**Reply by jsonnentag** [reply]
Well, if what you said before is true (that you are using a version you have modified since the one that is available for downloading) then that is the likely reason, because there is absolutely nothing special that I did to get those results.

I opened the program, imported the feed, and then double clicked it to view the imported feed.  After that, all that was done was double click a feed item and view the Date added being changed and not matching the Release date anymore.

** So if you import and open a feed with what you are using and you see the same dates for Release and Added what happens when you SAVE that episode/short form video?  (You will have to add a rating level and source or course.)**

When you look at the JSON entry does it change the Date added to the format you are using?  (So that when you look at it in a browser or some JSON file viewer it doesn't end with a "Z" anymore and doesn't list out 3 zeros after a decimal in the seconds position anymore and starts using the +00:00?)

I haven't played with it to see, but even if something starts in the format you are checking for does it leave the hours, minutes, and seconds as they were originally?  Because that is an important thing when you are sorting things when you post a series that was recorded and posted at different times during the same day (such as conference meetings or that type of stuff)!  The way things are now you don't even have the option to add hours, minutes, and seconds to the date added field in the Feed Manager.  That is probably the very reason that Roku sorts/sorted things by Date added and NOT by Release date.  Release date doesn't have any hours, minutes, and seconds for sorting things that happened on the same day.

**Reply by rrirower** [reply]
I would normally buy that explanation except that the version I am using does not affect the code dealing with 'dateAdded'.  I'll have to try and isolate this further.

**Comment by rrirower** [comment]
I've been able to isolate this issue.  It would occur only if I update something else on the screen like rating.  I should have a fix for it in the next release.

In the future, if you run into something you suspect may be a bug, please use the [Bug Report](https://github.com/rrirower/Channel-Feed-Manager/issues/new?assignees=rrirower&labels=&projects=&template=bug_report.md&title=) template when submitting an issue.  It makes it much easier to determine what's involved and how to go about debugging.

**Reply by jsonnentag** [reply]
I **NEVER** made **ANY** changes to **ANYTHING** when seeing the date changed!

I did exactly and ONLY what I listed in order to see that date NOT MATCH the date in the JSON file that was imported.

Just to make sure it isn't doing something when you "import" the JSON I deleted it just now in the Feed Manager and then added it again.  I went to the subdirectory and looked at the JSON that is copied in and it has the original matching dates in the JSON file.

Then I did EXACTLY the same thing I have always done and double clicked the feed in the "library" and then double clicked the first item visible as a video.  **ABSOLUTELY NOTHING** else is done.  The dates that are visible do not match.  **ZERO** else is done.  I open the feed and view a video's info and the daters don't match.  I never change a single thing before seeing the dates not matching!

Windows 10 with all updates and the Feed Manager downloaded straight from GitHub.

**Comment by rrirower** [comment]
Resolved in version 1.0.1.3
